### PR TITLE
client/http: implement more rule and batch related interfaces

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #7300.

### What is changed and how does it work?

```commit-message
- Implement more rule and batch related interfaces.
- Add more types and methods.
- Refine the tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/http/client.go [hunk 1]
```diff
@@ -446,6 +467,48 @@ func (c *client) DeletePlacementRule(ctx context.Context, group, id string) erro
 		http.MethodDelete, http.NoBody, nil)
 }
 
+// GetAllPlacementRuleGroups gets all placement rule groups.
+func (c *client) GetAllPlacementRuleGroups(ctx context.Context) ([]*RuleGroup, error) {
+	var ruleGroups []*RuleGroup
+	err := c.requestWithRetry(ctx,
+		"GetAllPlacementRuleGroups", placementRuleGroups,
+		http.MethodGet, nil, &ruleGroups)
```
Comment (HuSharp) [inline client/http/client.go:475]: maybe need to sync with this change
https://github.com/tikv/pd/pull/7428/files#diff-138f7d23a1171f829bb57a7edcd66e8ac0ceb81f1cb2f28e497284c732aaad22L339

Comment (JmPotato) [reply client/http/client.go]: Done. PTAL.

File: client/http/client.go [hunk 2]
```diff
@@ -427,6 +437,17 @@ func (c *client) SetPlacementRule(ctx context.Context, rule *Rule) error {
 		http.MethodPost, bytes.NewBuffer(ruleJSON), nil)
 }
 
+// SetPlacementRules sets the placement rules in batch.
```
Comment (HuSharp) [inline client/http/client.go:440]:
```suggestion
// SetPlacementRuleInBatch sets the placement rules in batch.
```

**Review by HuSharp — APPROVED** [review]
LGTM!

File: client/http/client.go [hunk 3]
```diff
@@ -57,14 +58,23 @@ type Client interface {
...
+	GetPlacementRuleGroupByID(context.Context, string) (*RuleGroup, error)
+	SetPlacementRuleGroup(context.Context, *RuleGroup) error
+	DeletePlacementRuleGroupByID(context.Context, string) error
 	GetAllRegionLabelRules(context.Context) ([]*LabelRule, error)
 	GetRegionLabelRulesByIDs(context.Context, []string) ([]*LabelRule, error)
 	SetRegionLabelRule(context.Context, *LabelRule) error
 	PatchRegionLabelRules(context.Context, *LabelRulePatch) error
 	/* Scheduling-related interfaces */
 	AccelerateSchedule(context.Context, []byte, []byte) error
+	AccelerateScheduleInBatch(context.Context, []struct {
```
Comment (rleungx) [inline client/http/client.go:74]: Shall we define a struct or use [][]byte?

**Review by rleungx — APPROVED** [review]
The rest LGTM.

**Pull request merged**